mono/metadata/appdomain.c

```diff
@@ -831,25 +838,35 @@ mono_class_get_appdomain_do_type_resolve_method (MonoError *error)
  *
  * \returns A \c MonoReflectionAssembly or NULL if not found
  */
-MonoReflectionAssembly *
+MonoReflectionAssemblyHandle
 mono_domain_try_type_resolve_name (MonoDomain *domain, const char *name, MonoError *error)
 {
+	HANDLE_FUNCTION_ENTER ();
+
 	g_assert (domain);
 	g_assert (name);
 	g_assert (error);
 
 	error_init (error);
 
-	void *params [] = { mono_string_new_checked (mono_domain_get (), name, error) };
-	return_val_if_nok (error, NULL);
-
-	MonoMethod * const method = mono_class_get_appdomain_do_type_resolve_method (error);
-	return_val_if_nok (error, NULL);
-
-	MonoObject * const ret = mono_runtime_invoke_checked (method, domain->domain, params, error);
-	return_val_if_nok (error, NULL);
+	void *params [ ] = { mono_string_new_checked (mono_domain_get (), name, error) };
```

Comment:
Move this closer to the invoke.  Or else alloc the string into a MonoStringHandle. Right now this is relying on no GC in all the calls in between.